pysph/solver/tools.py

```diff
@@ -151,9 +151,65 @@ def _get_sph_eval_shepard(self):
         else:
             return self._sph_eval
 
+    def _get_sph_eval_mls2d_1(self):
+        if self._sph_eval is None:
+            from pysph.sph.wc.density_correction import (
+                MLSFirstOrderPreStep2D, MLSFirstOrder)
+            from pysph.tools.sph_evaluator import SPHEvaluator
+            from pysph.sph.equation import Group
+            arrs = self.arrs
+            eqns = []
+            n = 3
+            for arr in arrs:
+                lenga = len(arr.x) * n * n
+                lengb = len(arr.x) * n
+                arr.add_constant('a', [0.] * lenga)
```

Comment:
This name is a bit too generic: 'a', 'b' , please use something a bit more meaningful.  Also are these really needed?  Do you really need to store the matrix for each particle?

Reply:
Maybe we can revisit this when we add support to handle all neighbors.